src/systems/filecoin_blockchain/chainsync/_index.md

```diff
@@ -345,22 +347,23 @@ State Machine:
+- It is key to note that any block received after the `EPOCH_CUTOFF` time must be automatically discarded by the miner until the start of the next epoch.
 
 - **Progressive Stages of Block Validation**
   - _(TODO: move this to blockchain/Block section)_
   - **BV0 - Syntactic Validation**: Validate data structure packing and ensure correct typing.
-  - **BV1 - Light Consensus State Checks**: Validate `b.ChainWeight`, `b.ChainEpoch`, `b.MinerAddress`, `b.Timestamp`, are plausible (some ranges of bad values can be detected easily, especially if we have the state of the chain at `b.ChainEpoch - consensus.LookbackParameter`. Eg Weight and Epoch have well defined valid ranges, and `b.MinerAddress`
-  must exist in the lookback state). This requires some chain state, enough to establish plausibility levels of each of these values. A node should be able to estimate valid ranges for `b.ChainEpoch` based on the `LastTrustedCheckpoint`. `b.ChainWeight` is easy if some of the relatively recent chain is available, otherwise hard.
-  - **BV2 - Signature Validation**: Verify `b.BlockSig` is correct.
-  - **BV3 - Verify ElectionProof**: Verify `b.ElectionProof` is correct. This requires having state for relevant lookback parameters.
-  - **BV4 - Verify Ancestry links to chain**: Verify ancestry links back to trusted blocks. If the ancestry forks off before finality, or does not connect at all, it is a bad block.
-  - **BV4 - Verify MessageSigs**: Verify the signatures on messages
-  - **BV5 - Verify StateTree**: Verify the application of `b.Parents.Messages()` correctly produces `b.StateTree` and `b.MessageReceipts`
+  - **BV1 - Epoch Window Validation**: Validate that block was produced during the current epoch.  Validate that the block was received before `EPOCH_CUTOFF`.  This stage only applies during `CHAIN_FOLLOW`.
+  - **BV2 - Light Consensus State Checks**: Validate `b.ChainWeight`, `b.ChainEpoch`, `b.MinerAddress`, `b.Timestamp`, are plausible (some ranges of bad values can be detected easily, especially if we have the state of the chain at `b.ChainEpoch - consensus.LookbackParameter`. Eg Weight and Epoch have well defined valid ranges, and `b.MinerAddress`
+  must exist in the lookback state). This requires some chain state, enough to establish plausibility levels of each of these values. A node can determine the exact valid range for `b.ChainEpoch` based on the `LastTrustedCheckpoint`, and a synchronized clock (see {{<sref clock>}}). `b.ChainWeight` is easy if some of the relatively recent chain is available, otherwise hard.
+  - **BV3 - Signature Validation**: Verify `b.BlockSig` is correct.
+  - **BV4 - Verify ElectionProof**: Verify `b.ElectionProof` is correct. This requires having state for relevant lookback parameters.
+  - **BV5 - Verify Ancestry links to chain**: Verify ancestry links back to trusted blocks. If the ancestry forks off before finality, or does not connect at all, it is a bad block.
```

Comment:
Does this fit within the `candidateHeads` and `validationFringe` model? My sense is your worked forward, not bacl.

Reply:
Everything past BV1 is just copy pasted from shifting the number, so idk. The spec's intent might be to sync forwards but currently I read it as leaving it up to implementation.  and go-filecoin for sure syncs backwards.

Reply:
I believe syncing backwards is correct -- nodes shouldn't waste resources on validating chains that might not lead to heads the network believes to be current.